crates/ruff_python_parser/src/lexer.rs

```diff
@@ -264,11 +264,29 @@ impl<'source> Lexer<'source> {
 
         let mut number = LexedText::new(self.offset(), self.source);
         self.radix_run(&mut number, radix);
-        let value =
-            BigInt::from_str_radix(number.as_str(), radix.as_u32()).map_err(|e| LexicalError {
-                error: LexicalErrorType::OtherError(format!("{e:?}")),
+        // 2**31 is 2147483648, i.e 10 digits, so all 9 digit numbers are representable as i32
+        // TODO: Can we go up to i64?
+        if radix == Radix::Decimal && number.as_str().len() < 10 {
```

Comment:
This method is never called with `Radix::Decimal`. You need to add your fast path to `lex_decimal_number`